# Import libraries

In [1]:
# drive access
from google.colab import drive
drive.mount('/content/drive')

# standard library
import numpy as np
import pandas as pd

Mounted at /content/drive


# Define reusable functions

In [2]:
def front_assertion(df):
  front_assertions = []

  for i, row in df.iterrows():
    if str(row['auto_transcription'])[:5].lower() == str(row['label'])[:5].lower():
      front_assertions.append('Y')
    else:
      front_assertions.append('N')

  assert len(front_assertions) == len(df)
  df['front_assertions'] = front_assertions

In [3]:
def back_assertion(df):
  back_assertions = []

  for i, row in df.iterrows():
    if str(row['auto_transcription'])[-5:].lower() == str(row['label']).replace('.','')[-5:].lower():
        back_assertions.append('Y')
    else:
      back_assertions.append('N')

  assert len(back_assertions) == len(df)
  df['back_assertions'] = back_assertions

# Clean EmoV_DB main.csv

## Read csv file

In [4]:
emov_df = pd.read_csv('/content/drive/MyDrive/266/Data/Clean_Data/EmoV_DB/main.csv')

emov_df.head()

filename                                 auto_transcription  \
0    amused_29-45_0042.wav  HOW COULD HE EXPLAIN HIS POSSESSION OF THE SKETCH   
1    amused_46-56_0046.wav  THE GIRL FACED HIM HER EYES SHINING WITH SUDDE...   
2     amused_1-15_0005.wav                             WILL WE EVER FORGET IT   
3  amused_281-308_0281.wav      I DO NOT BLAME YOU FOR ANYTHING REMEMBER THAT   
4  amused_225-252_0226.wav                         THAT CAME BEFORE MY A V CS   

  emotion actor  gender  clean_filename  \
0  amused   bea  female              42   
1  amused   bea  female              46   
2  amused   bea  female               5   
3  amused   bea  female             281   
4  amused   bea  female             226   

                                               label  
0  How could he explain his possession of the ske...  
1  The girl faced him, her eyes shining with sudd...  
2                            Will we ever forget it.  
3    I do not blame you for anything; remember that.  
4                       That came before my A B C's.

In [5]:
emov_df['emotion'].value_counts()

sleepy     1720
neutral    1568
amused     1317
angry      1268
disgust    1019
Name: emotion, dtype: int64

In [6]:
emov_df['actor'].value_counts()

sam      2453
jenie    1790
bea      1786
josh      863
Name: actor, dtype: int64

In [7]:
emov_df['gender'].value_counts()

female    3576
male      3316
Name: gender, dtype: int64

## Find instances of label with digits & replace with spelled out words

In [8]:
emov_df[emov_df['label'].str.contains(r'\d')]['label'].value_counts()

At sea, Wednesday, March 18, 1908.    9
At sea, Monday, March 16, 1908.       9
Name: label, dtype: int64

In [9]:
emov_df['label'] = emov_df['label'].str.replace('1908', 'nineteen o eight')
emov_df['label'] = emov_df['label'].str.replace('16', 'sixteenth')
emov_df['label'] = emov_df['label'].str.replace('18', 'eighteenth')

In [10]:
assert len(emov_df[emov_df['label'].str.contains(r'\d')]) == 0

## Inspect very noisy auto-transcription

In [11]:
front_assertion(emov_df)
back_assertion(emov_df)

emov_df.head()

filename                                 auto_transcription  \
0    amused_29-45_0042.wav  HOW COULD HE EXPLAIN HIS POSSESSION OF THE SKETCH   
1    amused_46-56_0046.wav  THE GIRL FACED HIM HER EYES SHINING WITH SUDDE...   
2     amused_1-15_0005.wav                             WILL WE EVER FORGET IT   
3  amused_281-308_0281.wav      I DO NOT BLAME YOU FOR ANYTHING REMEMBER THAT   
4  amused_225-252_0226.wav                         THAT CAME BEFORE MY A V CS   

  emotion actor  gender  clean_filename  \
0  amused   bea  female              42   
1  amused   bea  female              46   
2  amused   bea  female               5   
3  amused   bea  female             281   
4  amused   bea  female             226   

                                               label front_assertions  \
0  How could he explain his possession of the ske...                Y   
1  The girl faced him, her eyes shining with sudd...                Y   
2                            Will we ever forget it.                Y   
3    I do not blame you for anything; remember that.                Y   
4                       That came before my A B C's.                Y   

  back_assertions  
0               Y  
1               Y  
2               Y  
3               Y  
4               N

In [12]:
emov_failed = emov_df[(emov_df['front_assertions'] == 'N') & (emov_df['back_assertions'] == 'N')]

emov_failed

filename  \
25         amused_85-112_0110.wav   
71        amused_253-280_0253.wav   
84           amused_1-15_0004.wav   
92        amused_196-224_0206.wav   
94           amused_1-15_0015.wav   
...                           ...   
6882  sleepiness_393-420_0403.wav   
6884  sleepiness_393-420_0397.wav   
6885  sleepiness_421-448_0439.wav   
6887  sleepiness_477-504_0484.wav   
6888  sleepiness_449-476_0455.wav   

                                     auto_transcription emotion actor  gender  \
25                         MUCH REPLIED JEAN AS TERSLEY  amused   bea  female   
71    HE WAS THAT LIVED TO FOUND THE FAMILY OF THE P...  amused   bea  female   
84              LORD BUT I'M GLAD TO SEE YOU AGAIN VELL  amused   bea  female   
92            HAHA HE SAYS HE BOUGHT HIM OF JACK LEBULL  amused   bea  female   
94                   HA HA HA IT'S THE AURORA BOREALICE  amused   bea  female   
...                                                 ...     ...   ...     ...   
6882     IF YOU ARE CUNNING GAVE HIM POISE HEDTHE SHRAW  sleepy   sam    male   
6884          SOLL ME THEN AGAIN HE MURMURED ITSTILATLY  sleepy   sam    male   
6885           I'D SEE WHA SAY MARG EIGHTEENT LOTITO LA  sleepy   sam    male   
6887                                          NO CERENE  sleepy   sam    male   
6888  I HOPE BEY A EVILITE TOU FATER THEY SIT FAFAR ...  sleepy   sam    male   

      clean_filename                                              label  \
25               110                  Much, replied Jeanne, as tersely.   
71               253  He it was that lived to found the family of th...   
84                 4         Lord, but I'm glad to see you again, Phil.   
92               206          He says he bought him of Jacques Le Beau.   
94                15                          It's the aurora borealis.   
...              ...                                                ...   
6882             403    His newborn cunning gave him poise and control.   
6884             397      Call me that again, he murmured ecstatically.   
6885             439  At sea, Wednesday, March eighteenth, nineteen ...   
6887             484                                         No-sir-ee.   
6888             455  I obeyed, and a minute or two later they stood...   

     front_assertions back_assertions  
25                  N               N  
71                  N               N  
84                  N               N  
92                  N               N  
94                  N               N  
...               ...             ...  
6882                N               N  
6884                N               N  
6885                N               N  
6887                N               N  
6888                N               N  

[296 rows x 9 columns]

We can see that Josh and Sam's sleepy and amused sets have many instances of both front and back assertion failures, let's take a closer look to those instances

In [13]:
emov_failed_grouped = emov_failed[['actor','emotion']].groupby(['actor','emotion']).value_counts()

emov_failed_grouped

actor  emotion
bea    amused      8
       angry       2
       neutral     2
       sleepy     11
jenie  amused      1
       angry       6
       disgust     2
       neutral     3
       sleepy     10
josh   amused     48
       neutral     3
       sleepy     54
sam    amused     49
       angry       9
       disgust    15
       neutral    13
       sleepy     60
dtype: int64

There are 296 instances where both front and back assertion failed, around 4% of the total 6.9K instances

In [14]:
emov_failed_grouped.sum()

296

In [15]:
len(emov_df)

6892

In [16]:
emov_failed_grouped.sum() / len(emov_df)

0.042948345908299476

A good number of those failed instances for Josh and Sam Amused is due to 'speech laugh' which will be handled as part of Step 2 in the pipeline. The auto transcription is noisy but not unrecognizable, okay to leave as is.

In [17]:
emov_failed[(emov_failed['actor'] == 'josh') &  (emov_failed['emotion'] == 'amused')]

filename  \
3581  amused_225_252_0240.wav   
3582    amused_74_84_0079.wav   
3587   amused_85-112_0099.wav   
3590    amused_27-56_0031.wav   
3591   amused_85-112_0089.wav   
3594  amused_225_252_0247.wav   
3609     amused_1-28_0016.wav   
3610   amused_85-112_0103.wav   
3621  amused_281-308_0298.wav   
3622  amused_281-308_0304.wav   
3625  amused_252-280_0255.wav   
3642  amused_252-280_0265.wav   
3645  amused_113-140_0122.wav   
3651  amused_197-224_0198.wav   
3654   amused_85-112_0098.wav   
3661  amused_281-308_0299.wav   
3662     amused_1-28_0028.wav   
3663  amused_113-140_0138.wav   
3668     amused_1-28_0001.wav   
3677   amused_85-112_0093.wav   
3678     amused_1-28_0020.wav   
3687     amused_1-28_0024.wav   
3688  amused_225_252_0026.wav   
3704  amused_252-280_0277.wav   
3710  amused_225_252_0246.wav   
3716  amused_113-140_0120.wav   
3723     amused_1-28_0015.wav   
3726  amused_252-280_0253.wav   
3731    amused_27-56_0030.wav   
3735     amused_1-28_0022.wav   
3739  amused_197-224_0205.wav   
3742  amused_169-196_0195.wav   
3746  amused_169-196_0184.wav   
3750  amused_141_168_0152.wav   
3756  amused_169-196_0171.wav   
3775  amused_197-224_0214.wav   
3788  amused_197-224_0197.wav   
3789   amused_85-112_0100.wav   
3790  amused_281-308_0293.wav   
3794  amused_169-196_0190.wav   
3808  amused_252-280_0258.wav   
3837  amused_252-280_0267.wav   
3838    amused_27-56_0041.wav   
3839  amused_169-196_0192.wav   
3853    amused_27-56_0051.wav   
3854  amused_197-224_0222.wav   
3862    amused_27-56_0034.wav   
3869  amused_141_168_0168.wav   

                                     auto_transcription emotion actor gender  \
3581                         AA WASH YOUR HANDS WITH ME  amused  josh   male   
3582                                     ISA LORDGANOEL  amused  josh   male   
3587               AM I HAD NY JOY FOUND IT IN HIS VERY  amused  josh   male   
3590                         T IFTY HUNDRED YARDS APARK  amused  josh   male   
3591                      AN THAT CROWIS PRESIDENCS UPI  amused  josh   male   
3594                       THE ARIPASER SOMETIMES A DAY  amused  josh   male   
3609  IIS FOR CHURCHILL ARRIVELSTOV YOND THE VIRGO S...  amused  josh   male   
3610  HA HA HA HA THE CAME NO CAMS ON THE BUCK OF TH...  amused  josh   male   
3621  THE CHASCTHET PROMISE OF CONTINUED ACQUAINTANC...  amused  josh   male   
3622                   NCRATUFATED LASS OF A GLASS A HA  amused  josh   male   
3625  HE OUGHT TO HAVE OUTEGURTION  BE SURE OF THE H...  amused  josh   male   
3642  HEE THE DISCOVERY SEEMS TO HAVE BEEN VEN IN TH...  amused  josh   male   
3645  AA TWO YEARS AGO I GIVE UP CIVILIZATION FOR THISE  amused  josh   male   
3651          SOME TOPES WORE GENSYE FILLED WITH VISION  amused  josh   male   
3654                                         HISALICARY  amused  josh   male   
3661  A A A I SELL YOU I AM DISGUSTED WITH THIS ADVE...  amused  josh   male   
3662                               ROPERY PRIVARY FRAUD  amused  josh   male   
3663  AA IN THE FICTURE HE SLOW EACH MOMENT A GREATE...  amused  josh   male   
3668    AAUTHER AT THE DEGETRAL PILLIP STEALS ET CETERA  amused  josh   male   
3677             HERPEPOMENT IF HE CROUCHED AND LISTENS  amused  josh   male   
3678  HUFFS AND VULSA SAID HE'S SCIRED TO BE AN WIT ...  amused  josh   male   
3687  HIS MACKER FORTS ON THE NORTH WHICH CIEFLY HIN...  amused  josh   male   
3688                            I CAME BEFORE MY A B CS  amused  josh   male   
3704  HA HA HA WE'RE GAVE OUT A STIFLING APOISONOUS ...  amused  josh   male   
3710  A YOU HAVE HEARD ALWAYS HOW HE WAS THE LOVER O...  amused  josh   male   
3716   A GREAT FOOTFEELS LIKE THAT OF A TINESE DEBUTART  amused  josh   male   
3723                                       SEE OEVRILUS  amused  josh   male   
3726  HA HA HA HA HA HERE WAS THAT LIVITE FOUND THE ...  amused  josh   male   
3731                            HAT HIS FACE AND LENGTH  amused  josh   male   
373

In [18]:
emov_failed[(emov_failed['actor'] == 'sam') &  (emov_failed['emotion'] == 'amused')]

filename  \
4451  amused_113-140_0124.wav   
4453   amused_85-112_0097.wav   
4464  amused_477-504_0501.wav   
4465  amused_225-252_0234.wav   
4467  amused_421-448_0424.wav   
4470  amused_421-448_0422.wav   
4497  amused_477-504_0483.wav   
4511  amused_477-504_0478.wav   
4521  amused_421-448_0423.wav   
4525  amused_225-252_0251.wav   
4533  amused_225-252_0245.wav   
4542  amused_449-476_0473.wav   
4543  amused_225-252_0236.wav   
4551  amused_253-280_0253.wav   
4552  amused_449-476_0467.wav   
4583  amused_421-448_0442.wav   
4588  amused_367-392_0388.wav   
4589  amused_477-504_0503.wav   
4602  amused_336-364_0341.wav   
4611  amused_281-308_0301.wav   
4666  amused_225-252_0244.wav   
4685  amused_309-336_0309.wav   
4690  amused_367-392_0387.wav   
4708  amused_113-140_0120.wav   
4711  amused_477-504_0498.wav   
4718  amused_169-196_0175.wav   
4723  amused_253-280_0277.wav   
4727  amused_309-336_0315.wav   
4729    amused_57-84_0079.wav   
4733  amused_169-196_0195.wav   
4763  amused_477-504_0484.wav   
4777  amused_421-448_0444.wav   
4784  amused_169-196_0171.wav   
4785  amused_169-196_0188.wav   
4817  amused_169-196_0193.wav   
4835  amused_197-224_0205.wav   
4836  amused_393-420_0397.wav   
4847  amused_141-168_0143.wav   
4858  amused_393-420_0398.wav   
4860  amused_393-420_0404.wav   
4864  amused_393-420_0408.wav   
4867  amused_393-420_0400.wav   
4873  amused_141-168_0168.wav   
4879  amused_141-168_0158.wav   
4890  amused_393-420_0403.wav   
4894     amused_1-28_0001.wav   
4916  amused_393-420_0393.wav   
4921  amused_393-420_0417.wav   
4927  amused_393-420_0420.wav   

                                     auto_transcription emotion actor gender  \
4451       HE WAS GIEF SIMMING SOFTLY OVER SON THE ROTS  amused   sam   male   
4453  AND YOU CAN ARRANGE YOURSELF COMFORTABLY AMONG...  amused   sam   male   
4464  YOU JOKING ME SIR THE OTHER MANAGED CHARGE TIT...  amused   sam   male   
4465  WHY THE AVERAGER VIEW IS MORE NAUTIATING THAN ...  amused   sam   male   
4467  HOW HE IS LAING IT WAS A DISEASE HAS SYOULD BE...  amused   sam   male   
4470  HALF WAY AROUND THE TRACK ONE TOUH HE GOT INTO...  amused   sam   male   
4497  PE ON DISPUTE CORY HUTCHENSON AND MARRIED MABE...  amused   sam   male   
4511  ALSO SHE HAS FORPIDDEN THOS SMOKING THEIR PATE...  amused   sam   male   
4521   MAGVE HE WHEN HE RETURNS FROM A TRIP DOHATO LULU  amused   sam   male   
4525  I I MAY MANAGED TO FREIGHT O CARTO BACK HIS WHILE  amused   sam   male   
4533  HALA IS EIGHTEEN HUNDRED HHE LEED ASIDES SIXTE...  amused   sam   male   
4542                         UNA E WAS CAM AND SNOWY HE  amused   sam   male   
4543                          SOT YOU SEEIM I HATE YYOU  amused   sam   male   
4551  HE HAS WHO LIVED THE FOUND THE FAMILY  THE BITINA  amused   sam   male   
4552                      IT SO EARLY IN THE VOYAGE TOO  amused   sam   male   
4583           YOUHIVE ON AN INCOME WITH YOUR FATHER ER  amused   sam   male   
4588                             IT HAS MY IDEA TO A TA  amused   sam   male   
4589  HIS DEEDY BLACK EYES SAW PARTIANS WHERE OTHER ...  amused   sam   male   
4602                  WHY HE TA COTTAN YOU ALL SHE GAIN  amused   sam   male   
4611                     ALL MY DREAMS VIOLATED THIS LA  amused   sam   male   
4666      HE WHAS THE SOULT OF DEVOTION TO HIS EMPLAYRS  amused   sam   male   
4685                 WHIT LED CORDS SAID MISSUS MORTVER  amused   sam   male   
4690  BOB GROWING DISGUSTED TURNED THAT SUDDENLY AND...  amused   sam   male   
4708  MY WRITE FOOT HEELS LIKE THAT OF A CHINESE DEV...  amused   sam   male   
4711                       VOVAHINS WE ARE NOW VERY TAT  amused   sam   male   
4718  DHOWN THERE THE EARTH WHOS ALREADY SWELLING WI...  amused   sam   male   
4723  MACOEN FELLUS TYPLING POISON HIS ATTMOSPHERE I...  amused   sam   male   
4727  SATSIN WAITED FOR SHE KNEW A FRESH IDEA AD STA...  amused   sam   male   
4729                                HE WAS A LARGE

The sleepy instances are certainly more noisy than others, we can see some instances where the auto transcription is basically unrecognizable due to the actor yawning while speaking, especially at the beginning of the sentence. However, majority of them are still not 'too' noisy, and compared to ~10K total instances (EmoV_DB + Arctic), the very noisy ones are only a small fraction, propose to leave them as is.

In [19]:
emov_failed[(emov_failed['actor'] == 'josh') &  (emov_failed['emotion'] == 'sleepy')]

filename  \
4184  sleepiness_169-196_0195.wav   
4187    sleepiness_29-56_0056.wav   
4188    sleepiness_29-56_0039.wav   
4198  sleepiness_197-224_0200.wav   
4204  sleepiness_253_280_0265.wav   
4205  sleepiness_253_280_0267.wav   
4207  sleepiness_169-196_0192.wav   
4214  sleepiness_281-308_0308.wav   
4215   sleepiness_85-112_0095.wav   
4219  sleepiness_253_280_0270.wav   
4221  sleepiness_253_280_0253.wav   
4222  sleepiness_197-224_0215.wav   
4224  sleepiness_225-252_0251.wav   
4256  sleepiness_281-308_0292.wav   
4261  sleepiness_281-308_0296.wav   
4262  sleepiness_225-252_0234.wav   
4263  sleepiness_225-252_0238.wav   
4266  sleepiness_253_280_0254.wav   
4268  sleepiness_281-308_0291.wav   
4274  sleepiness_253_280_0277.wav   
4276  sleepiness_197-224_0217.wav   
4280  sleepiness_225-252_0225.wav   
4287  sleepiness_225-252_0248.wav   
4296  sleepiness_253_280_0274.wav   
4298  sleepiness_197-224_0216.wav   
4304   sleepiness_85-112_0099.wav   
4305    sleepiness_57-84_0078.wav   
4309  sleepiness_113_140_0138.wav   
4311   sleepiness_85-112_0086.wav   
4312  sleepiness_225-252_0232.wav   
4317  sleepiness_197-224_0211.wav   
4336  sleepiness_281-308_0305.wav   
4338    sleepiness_29-56_0047.wav   
4350  sleepiness_169-196_0171.wav   
4358  sleepiness_197-224_0224.wav   
4361  sleepiness_253_280_0276.wav   
4366   sleepiness_85-112_0110.wav   
4376  sleepiness_225-252_0252.wav   
4377  sleepiness_169-196_0177.wav   
4379  sleepiness_197-224_0207.wav   
4384  sleepiness_197-224_0208.wav   
4387  sleepiness_169-196_0186.wav   
4390  sleepiness_113_140_0139.wav   
4395  sleepiness_253_280_0266.wav   
4397  sleepiness_197-224_0210.wav   
4398  sleepiness_281-308_0299.wav   
4399  sleepiness_169-196_0188.wav   
4400  sleepiness_281-308_0285.wav   
4404  sleepiness_281-308_0283.wav   
4406  sleepiness_169-196_0187.wav   
4409  sleepiness_281-308_0287.wav   
4428  sleepiness_281-308_0307.wav   
4430  sleepiness_225-252_0241.wav   
4434  sleepiness_169-196_0184.wav   

                                     auto_transcription emotion actor gender  \
4184                              THE STRANGETI HAPPENS  sleepy  josh   male   
4187      PERS'S LITTLE EYES WERE FIXED ON HIM SREWEDLY  sleepy  josh   male   
4188       THEE SHIPS TO BE IN WITH IT A WEEK VERY INDY  sleepy  josh   male   
4198      HE LIVED AGAIN AND THE COAB GAUT A WOONCEMORE  sleepy  josh   male   
4204       THE SKOVER YE SEEM TO AVMEN NOW SPURROW OOLI  sleepy  josh   male   
4205         EI HARDING ASKED THE SHANK SAW HER TO FALL  sleepy  josh   male   
4207                                 HEDID NOT RUSH HIM  sleepy  josh   male   
4214          AS IF HER HERRY WER FEVEN KNOWING VACARLY  sleepy  josh   male   
4215  HER BIG GAVOUS TENT WAS THE FIRST THING TO COM...  sleepy  josh   male   
4219       AFTER THE BATH A SHAVE WOULD NOT BE SOBERRED  sleepy  josh   male   
4221    HER WAS THAT REBDIFEN THE FAMILY OF THE BPOTINO  sleepy  josh   male   
4222                           HAND HE DIRED OUT AF SPY  sleepy  josh   male   
4224                      I REBAY CHIFFREY CARQAPACISMI  sleepy  josh   male   
4256        HERE IN THE MIDWAY THE FIRST CASTLTY OHEARD  sleepy  josh   male   
4261                         RALSE WAS A FASERIOUS MILE  sleepy  josh   male   
4262  WHY THE AVERAGE REVIEW IS MORE NAUCIATING THAN...  sleepy  josh   male   
4263  HAPPENED TO HIM WIT THE GUI THIS IIN OKWANS WE...  sleepy  josh   male   
4266       SHADOW AY SWEAR THE HADS GROWING SMALL SMALL  sleepy  josh   male   
4268  THE REEKS HAD GONE BY AND NO OVERT AXE HAD BEE...  sleepy  josh   male   
4274        ACOII LAKLY HOS ITIS ATMOSPHERE IN THE PANK  sleepy  josh   male   
4276       A PRESSIESIHILB IT WAS NOW EVEN MORE PRASSES  sleepy  josh   male   
4280        I IS BROUGHT UP YOUR BROS GIRLS AND WHA PRO  sleepy  josh   male   
4287                           I HAPPEN SAD TOO ALL RIG  sleepy  josh   male   
4296        RAOU I LISTEN AGAIN TO THE TALE OF THE HOLS  sleepy  josh

In [20]:
emov_failed[(emov_failed['actor'] == 'sam') &  (emov_failed['emotion'] == 'sleepy')]

filename  \
6397   sleepiness_85-112_0110.wav   
6405  sleepiness_336-364_0355.wav   
6406  sleepiness_336-364_0341.wav   
6420  sleepiness_336-364_0347.wav   
6437  sleepiness_336-364_0346.wav   
6440  sleepiness_336-364_0363.wav   
6458  sleepiness_449-476_0467.wav   
6470  sleepiness_225-252_0226.wav   
6474  sleepiness_225-252_0246.wav   
6494  sleepiness_253-280_0258.wav   
6496  sleepiness_169_196_0180.wav   
6502  sleepiness_141-168_0168.wav   
6514  sleepiness_141-168_0150.wav   
6518  sleepiness_141-168_0158.wav   
6528  sleepiness_421-448_0422.wav   
6542  sleepiness_253-280_0267.wav   
6549  sleepiness_367-392_0387.wav   
6552  sleepiness_477-504_0502.wav   
6553  sleepiness_225-252_0225.wav   
6563  sleepiness_169_196_0183.wav   
6578  sleepiness_477-504_0478.wav   
6579  sleepiness_281-308_0290.wav   
6582  sleepiness_141-168_0157.wav   
6587  sleepiness_393-420_0405.wav   
6588  sleepiness_197-224_0221.wav   
6608  sleepiness_113-140_0138.wav   
6612  sleepiness_113-140_0120.wav   
6620  sleepiness_477-504_0503.wav   
6628  sleepiness_421-448_0435.wav   
6629  sleepiness_225-252_0233.wav   
6632     sleepiness_1-28_0015.wav   
6652  sleepiness_421-448_0432.wav   
6658  sleepiness_449-476_0456.wav   
6662  sleepiness_169_196_0176.wav   
6664  sleepiness_367-392_0383.wav   
6670  sleepiness_367-392_0369.wav   
6675  sleepiness_169_196_0172.wav   
6677    sleepiness_29-56_0056.wav   
6686  sleepiness_421-448_0423.wav   
6739  sleepiness_281-308_0387.wav   
6753  sleepiness_169_196_0181.wav   
6761    sleepiness_29-56_0049.wav   
6779  sleepiness_169_196_0188.wav   
6782  sleepiness_393-420_0401.wav   
6785  sleepiness_367-392_0375.wav   
6808  sleepiness_169_196_0192.wav   
6815  sleepiness_141-168_0160.wav   
6816  sleepiness_253-280_0261.wav   
6829  sleepiness_393-420_0417.wav   
6837  sleepiness_393-420_0404.wav   
6841  sleepiness_477-504_0483.wav   
6850  sleepiness_393-420_0400.wav   
6858  sleepiness_225-252_0234.wav   
6865  sleepiness_393-420_0393.wav   
6876  sleepiness_281-308_0295.wav   
6882  sleepiness_393-420_0403.wav   
6884  sleepiness_393-420_0397.wav   
6885  sleepiness_421-448_0439.wav   
6887  sleepiness_477-504_0484.wav   
6888  sleepiness_449-476_0455.wav   

                                     auto_transcription emotion actor gender  \
6397                       MUCH REPLIED JEAN AS TRUTHLY  sleepy   sam   male   
6405                   A PURSE OF LAUGHTER WITH HIS RUR  sleepy   sam   male   
6406                        WHY TARGON YOU ALL SAGETGAN  sleepy   sam   male   
6420  FER GREATLY DELIGHTED WOR THIRTY THE THAT IS B...  sleepy   sam   male   
6437                                  SUT DOWN AN TOVEN  sleepy   sam   male   
6440            HIS THOT IT HAD SO DESPATCH THE MERGIET  sleepy   sam   male   
6458                      AT SO EARLY IN THE VOYAGE TOO  sleepy   sam   male   
6470                             HATI BEFORE WY HE SEES  sleepy   sam   male   
6474  YOUVE HEARD AOYS HOW HE WAS THE LOVER OF THE P...  sleepy   sam   male   
6494         SOTON SEY WATER THEY ALL SET THE HEAT RACE  sleepy   sam   male   
6496                     INOT WHAT SER PRAISING GO OBEY  sleepy   sam   male   
6502      LO BUSH WOUL OBTAINED THE FACE E LUFH NO STIN  sleepy   sam   male   
6514                               VI PIERE YOU SOUT AT  sleepy   sam   male   
6518                                 WAS THAT LOR COULD  sleepy   sam   male   
6528  HALF WAY ROUT THE TRACK ONE DONKY CARIS HERE A...  sleepy   sam   male   
6542           E I HARDLY YESS AS SHUNK STARTED THE FUL  sleepy   sam   male   
6549  BOB DRAWAND DISTUSTED DON'T THAT SO AN I TEMPT...  sleepy   sam   male   
6552  ETHETA UNUSUAL HORAVDORABLE WAS SUFFICIENT TO ...  sleepy   sam   male   
6553  OH WHO IS BROUGHT U AWAY MOST CURLS AND WHOAI ...  sleepy   sam   male   
6563                        THE AIR WAS GROWING CHILLIN  sleepy   sam   male   
6578  ALSO SHE HAD FORBIDDEN THEM SMOKING THEIR PIPE...  sleepy   sam   male   
6579   

# Clean Arctic main.csv

## Read csv file

In [21]:
arctic_df = pd.read_csv('/content/drive/MyDrive/266/Data/Clean_Data/Arctic/main.csv')

arctic_df.head()

filename                                 auto_transcription  \
0  arctic_a0004.wav            LORD BUT I'M GLAD TO SEE YOU AGAIN PHIL   
1  arctic_a0050.wav  IN SPITE OF THEIR ABSURDITY THE WORDS AFFECTED...   
2  arctic_a0102.wav                             HE WILL FOLLOW US SOON   
3  arctic_a0072.wav                          BUT WHO WAS ILEN'S DOUBLE   
4  arctic_a0124.wav   IT WAS JEAN SINGING SOFTLY OVER BEYOND THE ROCKS   

   emotion actor gender clean_filename  \
0  neutral   bdl   male          a0004   
1  neutral   bdl   male          a0050   
2  neutral   bdl   male          a0102   
3  neutral   bdl   male          a0072   
4  neutral   bdl   male          a0124   

                                               label  
0         Lord, but I'm glad to see you again, Phil.  
1  In spite of their absurdity the words affected...  
2                            He will follow us soon.  
3                       But who was Eileen's double.  
4  It was Jeanne singing softly over beyond the r...

In [22]:
arctic_df['emotion'].value_counts()

neutral    4528
Name: emotion, dtype: int64

In [23]:
arctic_df['actor'].value_counts()

bdl    1132
clb    1132
rms    1132
slt    1132
Name: actor, dtype: int64

In [24]:
arctic_df['gender'].value_counts()

male      2264
female    2264
Name: gender, dtype: int64

## Find instances of label with digits & replace with spelled out words

In [25]:
arctic_df[arctic_df['label'].str.contains(r'\d')]['label'].value_counts()

At sea, Monday, March 16, 1908.       4
At sea, Wednesday, March 18, 1908.    4
The 29th very foggy.                  4
At sea, Tuesday, March 17, 1908.      4
Name: label, dtype: int64

In [26]:
arctic_df['label'] = arctic_df['label'].str.replace('1908', 'nineteen o eight')
arctic_df['label'] = arctic_df['label'].str.replace('29th', 'twenty ninth')
arctic_df['label'] = arctic_df['label'].str.replace('16', 'sixteenth')
arctic_df['label'] = arctic_df['label'].str.replace('17', 'seventeenth')
arctic_df['label'] = arctic_df['label'].str.replace('18', 'eighteenth')

In [27]:
assert len(arctic_df[arctic_df['label'].str.contains(r'\d')]) == 0

## Inspect very noisy auto-transcription

In [28]:
front_assertion(arctic_df)
back_assertion(arctic_df)

arctic_df.head()

filename                                 auto_transcription  \
0  arctic_a0004.wav            LORD BUT I'M GLAD TO SEE YOU AGAIN PHIL   
1  arctic_a0050.wav  IN SPITE OF THEIR ABSURDITY THE WORDS AFFECTED...   
2  arctic_a0102.wav                             HE WILL FOLLOW US SOON   
3  arctic_a0072.wav                          BUT WHO WAS ILEN'S DOUBLE   
4  arctic_a0124.wav   IT WAS JEAN SINGING SOFTLY OVER BEYOND THE ROCKS   

   emotion actor gender clean_filename  \
0  neutral   bdl   male          a0004   
1  neutral   bdl   male          a0050   
2  neutral   bdl   male          a0102   
3  neutral   bdl   male          a0072   
4  neutral   bdl   male          a0124   

                                               label front_assertions  \
0         Lord, but I'm glad to see you again, Phil.                N   
1  In spite of their absurdity the words affected...                Y   
2                            He will follow us soon.                Y   
3                       But who was Eileen's double.                Y   
4  It was Jeanne singing softly over beyond the r...                Y   

  back_assertions  
0               Y  
1               Y  
2               Y  
3               Y  
4               Y

In [29]:
arctic_failed = arctic_df[(arctic_df['front_assertions'] == 'N') & (arctic_df['back_assertions'] == 'N')]

arctic_failed

filename                                 auto_transcription  \
207   arctic_a0484.wav                                           NO SUREE   
236   arctic_a0478.wav  ALSO SHE HAS FORBIDDEN THEM SMOKING THEIR PIPE...   
299   arctic_a0423.wav  MAKE VAY WHEN HE RETURNS FROM A TRIP TO HANNALULU   
369   arctic_a0551.wav               ALL IN APPEARANCE CAN KNOW AS MARAGE   
416   arctic_a0506.wav  THE PLAUDIN WAS LEAVING NEXT MORNING FOR HANALULU   
442   arctic_a0523.wav  HEY COULD NOT CONTINUE THEIR METHOD OF PRODUCI...   
673   arctic_b0087.wav  HEY WILL SEARCH OR US BETWEEN THEIR CAMP AND C...   
675   arctic_b0156.wav  OR THAT REASON LIBEAU HAD CHOSEN HIM TO FIGHT ...   
1420  arctic_a0423.wav   MAC VAY WHEN HE RETURNS FROM A TRIP TO HANNALULU   
1477  arctic_a0478.wav  ALSO SHE HAS FORBIDDEN THEM SMOKING THEIR PIPE...   
1482  arctic_a0484.wav                                          NO SURREY   
1506  arctic_a0506.wav  THE KLAUDINE WAS LEAVING NEXT MORNING FOR HANA...   
2554  arctic_a0423.wav    MAC VEY WHEN HE RETURNS FROM A TRIP TO HANALULU   
2608  arctic_a0478.wav  ALSO SHE HAS FORBIDDEN THEM SMOKING THEIR PIPE...   
2614  arctic_a0484.wav                                            NO SURE   
3292  arctic_a0028.wav                              RIBBERY BRIBERY FRAUD   
3373  arctic_a0110.wav                       MUCH REPLIED GEAN ESTURSELEY   
3412  arctic_a0506.wav  THE KLAWDIN WAS LEAVING NEXT MORNING FOR HANNA...   
3605  arctic_a0478.wav  ALSO SHE HAS FORBIDDEN THEM SMOKING THEIR PIPE...   
4008  arctic_a0110.wav                      MUCH REPLIED JEAN AS TURSELEY   
4062  arctic_a0484.wav                                          NO SIR YE   
4163  arctic_a0423.wav   MC VAY WHEN HE RETURNS FROM A TRIP TO HUNNA LULU   
4447  arctic_a0551.wav               ALL IN APPEARANCE CAN KNOW IS MARAGE   

      emotion actor  gender clean_filename  \
207   neutral   bdl    male          a0484   
236   neutral   bdl    male          a0478   
299   neutral   bdl    male          a0423   
369   neutral   bdl    male          a0551   
416   neutral   bdl    male          a0506   
442   neutral   bdl    male          a0523   
673   neutral   bdl    male          b0087   
675   neutral   bdl    male          b0156   
1420  neutral   clb  female          a0423   
1477  neutral   clb  female          a0478   
1482  neutral   clb  female          a0484   
1506  neutral   clb  female          a0506   
2554  neutral   rms    male          a0423   
2608  neutral   rms    male          a0478   
2614  neutral   rms    male          a0484   
3292  neutral   rms    male          a0028   
3373  neutral   rms    male          a0110   
3412  neutral   slt  female          a0506   
3605  neutral   slt  female          a0478   
4008  neutral   slt  female          a0110   
4062  neutral   slt  female          a0484   
4163  neutral   slt  female          a0423   
4447  neutral   slt  female          a0551   

                                                  label front_assertions  \
207                                          No-sir-ee.                N   
236   Also, she has forbidden them smoking their pip...                N   
299    McVeigh when he returns from a trip to Honolulu.                N   
369               All an appearance can know is mirage.                N   
416   The Claudine was leaving next morning for Hono...                N   
442   They could not continue their method of produc...                N   
673   They will search for us between their camp and...                N   
675   For that reason Le Beau had chosen him to figh...                N   
1420   McVeigh when he returns from a trip to Honolulu.                N   
1477  Also, she has forbidden them smoking their pip...                N   
1482                                         No-sir-ee.                N   
1506  The Claudine was leaving next morning for Hono...                N   
2554   McVeigh when he returns from a trip to Honolulu.                N

We can see that the dataset is fairly clean without many noisy instances

In [30]:
arctic_failed_grouped = arctic_failed[['actor','emotion']].groupby(['actor','emotion']).value_counts()

arctic_failed_grouped

actor  emotion
bdl    neutral    8
clb    neutral    4
rms    neutral    5
slt    neutral    6
dtype: int64

There are 23 instances where both front and back assertion failed, around 0.5% of the total 4.5K instances

In [31]:
arctic_failed_grouped.sum()

23

In [32]:
len(arctic_df)

4528

In [33]:
arctic_failed_grouped.sum() / len(arctic_df)

0.0050795053003533566

# Merge EmoV_DB and Arctic to one dataset

In [34]:
assert emov_df.columns.equals(arctic_df.columns)

In [35]:
merged_df = pd.concat([emov_df, arctic_df])

merged_df.head()

filename                                 auto_transcription  \
0    amused_29-45_0042.wav  HOW COULD HE EXPLAIN HIS POSSESSION OF THE SKETCH   
1    amused_46-56_0046.wav  THE GIRL FACED HIM HER EYES SHINING WITH SUDDE...   
2     amused_1-15_0005.wav                             WILL WE EVER FORGET IT   
3  amused_281-308_0281.wav      I DO NOT BLAME YOU FOR ANYTHING REMEMBER THAT   
4  amused_225-252_0226.wav                         THAT CAME BEFORE MY A V CS   

  emotion actor  gender clean_filename  \
0  amused   bea  female             42   
1  amused   bea  female             46   
2  amused   bea  female              5   
3  amused   bea  female            281   
4  amused   bea  female            226   

                                               label front_assertions  \
0  How could he explain his possession of the ske...                Y   
1  The girl faced him, her eyes shining with sudd...                Y   
2                            Will we ever forget it.                Y   
3    I do not blame you for anything; remember that.                Y   
4                       That came before my A B C's.                Y   

  back_assertions  
0               Y  
1               Y  
2               Y  
3               Y  
4               N

In [36]:
len(merged_df)

11420

In [37]:
# drop the assertion columns
merged_df.drop(columns=['front_assertions', 'back_assertions'], inplace=True)

merged_df.head()

filename                                 auto_transcription  \
0    amused_29-45_0042.wav  HOW COULD HE EXPLAIN HIS POSSESSION OF THE SKETCH   
1    amused_46-56_0046.wav  THE GIRL FACED HIM HER EYES SHINING WITH SUDDE...   
2     amused_1-15_0005.wav                             WILL WE EVER FORGET IT   
3  amused_281-308_0281.wav      I DO NOT BLAME YOU FOR ANYTHING REMEMBER THAT   
4  amused_225-252_0226.wav                         THAT CAME BEFORE MY A V CS   

  emotion actor  gender clean_filename  \
0  amused   bea  female             42   
1  amused   bea  female             46   
2  amused   bea  female              5   
3  amused   bea  female            281   
4  amused   bea  female            226   

                                               label  
0  How could he explain his possession of the ske...  
1  The girl faced him, her eyes shining with sudd...  
2                            Will we ever forget it.  
3    I do not blame you for anything; remember that.  
4                       That came before my A B C's.

In [38]:
# clean-up the orders of the columns
ordered_cols = ['filename','clean_filename','actor','gender','emotion','auto_transcription','label']

merged_df = merged_df[ordered_cols]

merged_df.head()

filename clean_filename actor  gender emotion  \
0    amused_29-45_0042.wav             42   bea  female  amused   
1    amused_46-56_0046.wav             46   bea  female  amused   
2     amused_1-15_0005.wav              5   bea  female  amused   
3  amused_281-308_0281.wav            281   bea  female  amused   
4  amused_225-252_0226.wav            226   bea  female  amused   

                                  auto_transcription  \
0  HOW COULD HE EXPLAIN HIS POSSESSION OF THE SKETCH   
1  THE GIRL FACED HIM HER EYES SHINING WITH SUDDE...   
2                             WILL WE EVER FORGET IT   
3      I DO NOT BLAME YOU FOR ANYTHING REMEMBER THAT   
4                         THAT CAME BEFORE MY A V CS   

                                               label  
0  How could he explain his possession of the ske...  
1  The girl faced him, her eyes shining with sudd...  
2                            Will we ever forget it.  
3    I do not blame you for anything; remember that.  
4                       That came before my A B C's.

In [39]:
merged_df['emotion'].value_counts()

neutral    6096
sleepy     1720
amused     1317
angry      1268
disgust    1019
Name: emotion, dtype: int64

In [40]:
merged_df['gender'].value_counts()

female    5840
male      5580
Name: gender, dtype: int64

In [41]:
merged_df['label'].value_counts()

How could he explain his possession of the sketch.                       22
In spite of their absurdity the words affected Philip curiously.         22
Her own betrayal of herself was like tonic to Philip.                    22
I had faith in them.                                                     22
They were three hundred yards apart.                                     22
                                                                         ..
They laughed like two happy children.                                     4
Besides, had he not whipped the big owl in the forest.                    4
You're a devil for fighting, and will surely win.                         4
One guess will do, Ernest retorted.                                       4
It would give me nervous prostration. She said with chattering teeth.     1
Name: label, Length: 1133, dtype: int64

# Export the merged csv to disk

In [42]:
merged_df.to_csv('/content/drive/MyDrive/266/Data/Clean_Data/EmoV_Arctic/merged.csv', index=False)